packages/react/src/component-tests/IcPopoverMenu/IcPopoverMenu.cy.tsx

```diff
@@ -0,0 +1,107 @@
+/// <reference types="Cypress" />
+
+import React from "react";
+import { mount } from "cypress/react";
+import { HAVE_BEEN_CALLED_ONCE, NOT_BE_CALLED_ONCE } from "../utils/constants";
+import { setThresholdBasedOnEnv } from "../../../cypress/utils/helpers";
+import {
+  DisableButton,
+  DisabledPopoverMenu,
+  PopoverDropdown,
+  PopoverMenuDescription,
+  PopoverMenuWithVariants,
+  PopoverMenulinks,
+  PopoverWithMenuGroups,
+} from "./IcPopoverMenuData";
+
+const WIN_CONSOLE_SPY = "@spyWinConsoleLog";
+
+describe("IcPagination e2e, A11y and visual regression tests", () => {
+  beforeEach(() => {
+    cy.injectAxe();
+  });
+
+  afterEach(() => {
+    cy.task("generateReport");
+  });
+
+  it("default PopoverMenu", () => {
+    mount(<PopoverDropdown></PopoverDropdown>);
+    cy.checkHydrated("ic-popover-menu");
+    cy.compareSnapshot({
+      name: "default_beforeclick_button",
+      testThreshold: setThresholdBasedOnEnv(0.02),
+    });
+    cy.get("ic-button").click();
+    cy.compareSnapshot({
+      name: "default_afterclick_button",
+      testThreshold: setThresholdBasedOnEnv(0.02),
+    });
+  });
+
+  it("render PopoverMenu With Description", () => {
+    mount(<PopoverMenuDescription></PopoverMenuDescription>);
+    cy.get("ic-button").click();
+    cy.compareSnapshot({
+      name: "popovermenu_with_description",
+      testThreshold: setThresholdBasedOnEnv(0.02),
+    });
+  });
+
+  it("should not be clickable when disabled option", () => {
+    cy.spy(window.console, "log").as("spyWinConsoleLog");
+    mount(<DisabledPopoverMenu onClick={() => console.log()} />);
+    cy.checkHydrated("ic-popover-menu");
+    cy.get("ic-button").click();
+    cy.get("ic-menu-item").eq(0).click({ force: true });
+    cy.get("ic-menu-item").eq(1).click({ force: true });
+    cy.get("ic-menu-item").eq(2).click({ force: true });
+    cy.compareSnapshot({
+      name: "disabled_dropdown",
+      testThreshold: setThresholdBasedOnEnv(0.02),
+    });
+    cy.get(WIN_CONSOLE_SPY).should(NOT_BE_CALLED_ONCE);
+  });
+
+  it("should not be clickable when disabled button", () => {
+    cy.spy(window.console, "log").as("spyWinConsoleLog");
+    mount(<DisableButton onClick={() => console.log()} />);
+    cy.checkHydrated("ic-popover-menu");
+    cy.get("ic-button").click({ force: true });
+    cy.compareSnapshot({
+      name: "disabled_button",
+      testThreshold: setThresholdBasedOnEnv(0.02),
+    });
+    cy.get(WIN_CONSOLE_SPY).should(NOT_BE_CALLED_ONCE);
+  });
```

Comment:
I'm not sure this test is needed as it's only testing disabled button which should be covered by the button tests and not popover